src/psyclone/psyir/nodes/dynamic_omp_task_directive.py

```diff
@@ -0,0 +1,1998 @@
+        # do i = x, y, 32
+        # and we have code inside the task region which does
+        # for j = i, i + 32, 1
+        # my_temp_var = j+1
+        # array(my_temp_var) = ...
+        # In this case, we need to have this variable as an extra copy of
+        # the proxy parent loop variable. Additionally, there can be multiple
+        # value set for this variable, so we need to store all possible ones
+        # (as they could occur inside an if/else statement and both values)
+        # be visible for each set.
+        added = False
+        key_list = list(self._proxy_loop_vars.keys())
+        for ref in references:
+            if isinstance(ref.parent, ArrayMixin):
+                continue
+            for index, proxy_var in enumerate(key_list):
+                if ref.symbol == proxy_var:
+                    if lhs.symbol in self._proxy_loop_vars:
+                        if (
+                            rhs
+                            not in self._proxy_loop_vars[lhs.symbol][
+                                "parent_node"
+                            ]
+                        ):
+                            self._proxy_loop_vars[lhs.symbol][
+                                "parent_node"
+                            ].append(rhs.copy())
+                    else:
+                        subdict = {}
+                        subdict["parent_var"] = \
+                            self._proxy_loop_vars[proxy_var]["parent_var"]
+                        subdict["parent_node"] = [rhs.copy()]
+                        subdict["loop"] = node
+                        subdict["parent_loop"] = self._parent_loops[index]
+                        self._parent_loops.append(self._parent_loops[index])
+                        self._proxy_loop_vars[lhs.symbol] = subdict
+                    added = True
+            if added:
+                break
+
+        for ref in references:
+            self._evaluate_readonly_reference(
+                ref, private_list, firstprivate_list, shared_list, in_list
+            )
+
+    def _evaluate_loop(
+        self,
+        node,
+        private_list,
+        firstprivate_list,
+        shared_list,
+        in_list,
+        out_list,
+    ):
+        """
+        Evaluates a Loop node within this task Region. This is done in several
+        steps:
+        1. Check the loop variable, start/stop/step values, and ensure that
+           they are valid. The loop variable must not be shared and the start,
+           stop, and step variables are not ArrayReferences. It also detects if
+           the loop variable is a "proxy", i.e. it represents a parent loop's
+           variable as a chunked loop variable. Any variables that are not yet
+           private, firstprivate or shared will be declared as firstprivate (as
+           they are read before being accessed elsewhere).
+        2. Loop through each of the nodes in the Loop's Schedule child, and
```

Comment:
It's this that makes me think of a Visitor...

Reply:
As previous.